assets/js/vue-apps/form-components/budget-input.vue

```diff
@@ -20,16 +21,27 @@ export default {
     },
     data() {
         // Add a ready-to-use new row if the budget isn't over the limit already
-        const initialBudgetRows =
-            this.budgetData.length < this.maxItems
-                ? concat(this.budgetData, [{ item: '', cost: '' }])
-                : this.budgetData;
-
+        const shouldAddNewRow = () => {
```

Comment:
We should probably move this to `methods` and then add a unit test for this as there's a fair bit of logic going on in here now.